### ISSUE TITLE
Wagtail 2.7

### ISSUE THREAD

**Issue by thaxy** [squareweave/wagtailcolumnblocks#4]
Hi, I used this package in an old project.
I am wondering if this still works with the new StreamField design introduced in v2.7.

**Comment by stefan-squareweave** [comment]
Hello, we haven't tried it with Wagtail 2.7. My guess initially is that it won't.

**Comment by mabdullahabid** [comment]
Hi Stefan,

It was working perfectly with v2.6.3 but when I upgraded to v2.7 the columns are not rendered properly.

They're not rendered horizontally (side-by-side) instead they're on top of each other. I'm guessing this is due to the new wagtail-react-streamfield code which has been pushed to v2.7.

I've been trying to dissect the code and tweak the block_form code to make it work with v2.7 but it's been unsuccessful so far.

Do you have a timeline on when you can push out a version compatible with v2.7?

Thanks!

**Comment by stefan-squareweave** [comment]
@mabdullahabid It's unlikely we'd look at this anytime soon; we'd only look at moving our existing Wagtail products to 2.7 when we're forced to by a lack of security fixes on the 2.6.x branch. Once that happens though we'd have to look at addressing this issue.

**Comment by thaxy** [comment]
You can try to override to add some [custom ``css``](https://docs.wagtail.io/en/v2.7/advanced_topics/customisation/admin_templates.html) to the wagtail admin page to make this package work again:
```
{% extends "wagtailadmin/base.html" %}

{# adjust wagtail admin to load styles to make wagtailcolumnblocks compatible with Wagtail 2.7 stream field #}

{% load static %}

{% block css %}
{{ block.super }}
<link rel="stylesheet" href="{% static 'css/custom-wagtail-columns.css' %}" />
{% endblock %}
```

``custom-wagtail-columns.css``:
```
.columns-block .fields {
    display: flex;
}

.columns-block .fields li label:first-child {
    display: block;
    margin-bottom: 7.5px;
    float: unset;
    width: unset;
    padding: 0;
}

.columns-block .fields li {
    padding-right: 1.5em
}

.columns-block .fields li:last-child {
    padding-right: 0
}

#wagtail-columns-container {
    display: flex;
}

.flex-col-6 {
    flex: 6;
}

.flex-col-4 {
    flex: 4;
}

.flex-col-8 {
    flex: 8;
}
```

Then override ``templates\block_forms\columnsblock.html``:
```
{% load wagtailcore_tags %}

{% load static %}

<link rel="stylesheet" href="{% static 'css/custom-wagtail-columns.css' %}" />

<div id="wagtail-columns-container">
        {% for column, width in columns %}
        {# width is relative to a column grid width, e.g. 12 #}
        <div class="column flex-col-{{ width }}">
            {% include_block column %}
        </div>
        {% endfor %}
</div>
```

**Comment by stefan-squareweave** [comment]
This was fixed by #7 . Closing.
